otter/convergence/transforming.py

```diff
@@ -0,0 +1,117 @@
+"""
+Functions for transforming collections of steps.
+
+There are a few classes of transformations here:
+
+- optimizing (converting multiple steps into one for the purposes of reducing
+  API roundtrips)
+- limiting (by truncating the number of steps we take)
+- throttling (by wrapping multiple steps with one that gradually executes them)
+"""
```

Comment:
@manishtomar To jump on the bikeshedding bandwagon, I agree with you that separating based on type may not be a good idea, but I kind of think of optimizing, limiting, and throttling as a role in that shaves things down based on steps produced by the planner.  The planner produces steps based on the current state of the world, and this module refines those steps to be more optimized in terms of networking.

But I can also see the perspective that optimizing is a separate role from limiting and throttling.